src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java

```diff
@@ -583,6 +577,27 @@ else if(finalAmsVal.equalsIgnoreCase("roster"))
                 }).log("${header.amsURL},${header.finalAmsVal}")
                 .removeHeaders("*")
                 .toD("${header.amsURL}/api/v1/paybill/validate/${header.finalAmsVal}?bridgeEndpoint=true");
+
+        from("rest:POST:/api/v1/gsma/transaction")
+                .id("gsma-transfer")
+                .unmarshal().json(JsonLibrary.Jackson, GsmaTransfer.class)
+                .log(LoggingLevel.INFO,"GSMA Transfer Body")
+                .process(e->{
+                    GsmaTransfer gsmaTranfer=e.getIn().getBody(GsmaTransfer.class);
+                    logger.info("Gsma Transfer DTO : {}",String.valueOf(gsmaTranfer));
+                    String subtype=gsmaTranfer.getSubType();
+                    String type=gsmaTranfer.getType();
+                    String amsName= e.getIn().getHeader("amsName").toString();
+                    String accountHoldingInstitutionId=e.getIn().getHeader("accountHoldingInstitutionId").toString();
+                    // inbound-transfer-mifos-lion
+                    Map<String, Object> variables = new HashMap<>();
+                    variables.put(TENANT_ID,accountHoldingInstitutionId);
+                    variables.put(CHANNEL_REQUEST, objectMapper.writeValueAsString(gsmaTranfer));
+                    String workflowName=subtype+"_"+type+"-"+amsName+"-"+accountHoldingInstitutionId;
```

Comment:
can you use new Stringbuilder().append(subtype).append(SEPARATOR).....toString()